Reduce size of response  AIO-720

tbrick855 opened the pull request:
Use _sourceIncludes and _sourceExcludes to remove all websites but calling one.
Use hardcoded list ansFields = [
    'canonical_url',
    'canonical_website',
    'content_elements',
    'created_date',
    'credits',
    'description',
    'display_date',
    'duration',
    'first_publish_date',
    'headlines',
    'last_updated_date',
    'promo_image',
    'promo_items',
    'publish_date',
    'streams',
    'subheadlines',
    'subtitles',
    'subtype',
    'taxonomy.primary_section',
    'taxonomy.seo_keywords',
    'taxonomy.tags',
    'type',
    'video_type',
  ]
Add website.{arc-site} to only include calling website
remove fields in sourceExcludes from ansFields
Add fields in sourceIncludes from ansFields
use transform to move website.*.website_section to taxonomy.sections
add website_url and website that are missing from use of sourceInclude

tbrick855 committed:
Reduce size of response  AIO-720

Use _sourceIncludes and _sourceExcludes to remove all websites but call
one.
Use hardcoded list ansFields = [
    'canonical_url',
    'canonical_website',
    'content_elements',
    'created_date',
    'credits',
    'description',
    'display_date',
    'duration',
    'first_publish_date',
    'headlines',
    'last_updated_date',
    'promo_image',
    'promo_items',
    'publish_date',
    'streams',
    'subheadlines',
    'subtitles',
    'subtype',
    'taxonomy.primary_section',
    'taxonomy.seo_keywords',
    'taxonomy.tags',
    'type',
    'video_type',
  ]
Add website.{arc-site} to only include callingg website
remove fields in sourceExcludes from ansFields
Add fields in sourceIncludes from ansFields
use transform to move website.*.website_section to taxonomy.sections
add website_url and website that are missing from use of sourceInclude

tbrick855 committed:
Add changeset

stevethedev commented on blocks/feeds-source-collections-block/sources/collections.js:
```diff
@@ -51,8 +70,22 @@ const fetch = async (key = {}) => {
-  // ids results does not include content_elements unless specified in included_fields
-  const idsIncludedFields = includedFields || defaultIncludedFields
+  // limit C-API response to just this websites sections to reduce size
+  ansFields.push(`websites.${site}`)
```
If `ansFields` is a global variable when it's deployed, then this could create a periodic and difficult-to-debug problem on AWS Lambda. 

AWS Lambda sometimes reuses their internal VMs; so modifying a global variable can cause one user's request to affect the next user's request. Since we have little control over how AWS spins down these internal VMs, this can propagate errors that are difficult to track down and fix.

stevethedev replied:
If `ansFields` is a global variable when it's deployed, then this could create a periodic and difficult-to-debug problem on AWS Lambda.

AWS Lambda sometimes reuses their internal VMs; so modifying a global variable can cause one user's request to affect the next user's request. Since we have little control over how AWS spins down these internal VMs, this can propagate errors that are difficult to track down and fix.

Since I'm not well acquainted with this code, there might be some technical reason why this is not the case; e.g., the output is all wrapped in a function that causes `ansFields` to actually _not_ be a global variable, then that could be a non-problem.

stevethedev commented on blocks/feeds-source-collections-block/sources/collections.js:
```diff
@@ -51,8 +70,22 @@ const fetch = async (key = {}) => {
+  if (excludeFields) {
+    excludeFields.split(',').forEach((i) => {
+      if (i && ansFields.indexOf(i) !== -1) {
+        ansFields.splice(ansFields.indexOf(i), 1)
```
Same potential-statefulness issue as above.

stevethedev commented on blocks/feeds-source-collections-block/sources/collections.js:
```diff
@@ -51,8 +70,22 @@ const fetch = async (key = {}) => {
+  if (includeFields) {
+    includeFields
+      .split(',')
+      .forEach((i) => i && !ansFields.includes(i) && ansFields.push(i))
```
Same potential-statefulness issue as above.

tbrick855 committed:
Refactored based on PR feedback

stevethedev reviewed (commented):
Overall, I think this all looks good. I'm going to give @malavikakoppula a chance to look at this, though, just in case I've overlooked something important.

tbrick855 committed:
Create content-source utils

This needs to be published first before these packages can be used in
the blocks.

tbrick855 committed:
Merge pull request #527 from WPMedia/AIO-720_util

Aio 720 util
